src/main/java/com/hazelcast/jdbc/HazelcastJdbcStatement.java

```
@@ -0,0 +1,377 @@
+package com.hazelcast.jdbc;
+
+import com.hazelcast.internal.util.Preconditions;
+import com.hazelcast.sql.SqlResult;
+import com.hazelcast.sql.SqlStatement;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.sql.SQLWarning;
+import java.sql.Statement;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Collections;
+import java.util.List;
+
+class HazelcastJdbcStatement implements Statement {
+
+    /** Query timeout. */
+    private Duration queryTimeout = Duration.ZERO;
+
+    /** Fetch size. */
+    private int fetchSize;
+
+    /** Whether the statement is closed. */
+    private boolean closed;
+
+    /** Poolable flag. */
+    private boolean poolable;
+
+    /** Fetch direction hint. */
+    private int fetchDirection = ResultSet.FETCH_FORWARD;
+
+    /** Whether to close the statement when the result set is closed. */
+    private boolean closeOnCompletion;
+
+    /** Current result as an update count */
+    int updateCount = -1;
+
+    /** Current result as a result set */
+    ResultSet resultSet;
+
+    private final HazelcastJdbcClient client;
+    private final Connection connection;
+
+    HazelcastJdbcStatement(HazelcastJdbcClient client, Connection connection) {
+        this.client = client;
+        this.connection = connection;
+    }
+
+    @Override
+    public ResultSet executeQuery(String sql) throws SQLException {
```

The `executeQuery` method should throw an exception if the produced result is not a `result set`, or if it is called on an instance of `PreparedStatement`.